tests/conftest.py

```diff
@@ -95,21 +96,22 @@ def pytest_configure(config):
 
 def pytest_unconfigure(config):
     """Stop queue listener safely during pytest teardown."""
-    global _log_listener, _log_queue # Ensure _log_queue is global
+    global _log_listener, _log_queue
     if _log_listener:
         try:
-            # Explicitly drain the queue before stopping the listener.
-            # This processes all remaining messages and prevents the I/O error.
+            # Drain remaining messages safely
             while not _log_queue.empty():
-                 # Use a non-blocking get (timeout=0)
-                 _log_listener.handle(_log_queue.get(timeout=0))
-            
+                try:
+                    record = _log_queue.get_nowait()
+                    _log_listener.handle(record)
+                except queue.Empty:
+                    break
             _log_listener.stop()
         except Exception:
-            # Suppress any errors caused by closed handlers during final stop
             pass
```

Comment:
This might be quite dangerous. It can completely hide valid exceptions that should propagate the exception chain.

Can we a bit more specific about what we need to catch here?